frontend/lib/Views/AdminPage/users/edit_user_roles_dialog.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:adpv_frontend/Models/all_users_list_provider.dart';
+import 'package:adpv_frontend/Repository/AdminRepository/users_list_repository.dart';
+import 'package:adpv_frontend/Views/AdminPage/confirmation_dialog_modal.dart';
+import 'package:flutter/material.dart';
+
+import '../groups/group_endpoint_view.dart';
+import '../utils.dart';
+
+const List<String> roles = ["USER", "ADMIN"];
+
+void editUserRoleDialog(
+  BuildContext context,
+  UserListData userListData,
+  AllUsersListProvider provider,
+) {
+  showDialog(
+    context: context,
+    builder: (context) {
+      final List<String> toSelect = List.from(userListData.roles);
+      return StatefulBuilder(
+        builder: (context, setState) => AlertDialog(
+          title: Text(
+            "Choose user roles:",
+            style: defaultAdminTextStyle,
+          ),
+          contentPadding: EdgeInsets.zero,
+          content: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              _buildRolesListTiles(toSelect, setState),
+              Padding(
+                padding: const EdgeInsets.symmetric(vertical: 10),
+                child: _buildButtons(context, userListData, provider, toSelect),
+              )
+            ],
+          ),
+        ),
+      );
+    },
+  );
+}
+
+Column _buildRolesListTiles(List<String> toSelect, StateSetter setState) =>
+    Column(
+      mainAxisSize: MainAxisSize.min,
+      children: roles
+          .map(
+            (element) => ListTile(
+              title: Text(
+                element,
+                style: defaultAdminTextStyle,
+              ),
+              leading: Checkbox(
+                shape: roundedRectangleBorder,
+                activeColor: Colors.teal,
+                value: toSelect.contains(element),
+                onChanged: (bool? value) {
+                  setState(() {
+                    if (value != null && value) {
```

Comment:
```suggestion
                    if (value == true) {
```

Reply:
Jedyne co można tu uprościć to warunek zamienić na `  if (value ?? true)`. Flutter tak nie działa.

Reply:
Okej, myślałem, że jak porównamy, czy dana zmienna jest `true` to wystarczy, bo w innych językach to działa